Wordsmithing return by ref (#1006)
dotnet/csharpstandard: standard/statements.md

A `return` statement is executed as follows:

> - For a return-by-value, *expression* is evaluated and its value is converted to the effective return type of the containing function by an implicit conversion. The result of the conversion becomes the result value produced by the function. For a return-by-ref, the *expression* is evaluated the same, but the result must be a reference to a *variable_reference* designated by *expression*. If the enclosing method’s return-by-ref includes `readonly`, the resulting variable is read-only.

Reviewer: ```suggestion
- For a return-by-value, *expression* is evaluated and its value is converted to the effective return type of the containing function by an implicit conversion. The result of the conversion becomes the result value produced by the function. For a return-by-ref, the *expression* is evaluated, and the result shall be classified as a variable. If the enclosing method’s return-by-ref includes `readonly`, the resulting variable is read-only.
```

> - For a return-by-value, *expression* is evaluated and its value is converted to the effective return type of the containing function by an implicit conversion. The result of the conversion becomes the result value produced by the function. For a return-by-ref, the *expression* is evaluated the same, but the result must be a reference to a *variable_reference* designated by *expression*. If the enclosing method’s return-by-ref includes `readonly`, the resulting variable is read-only.

Reviewer: Is "reference to a variable_reference" correct? And it's not clear how "designated by *expression*" comes into this... possibly remove?

> - For a return-by-value, *expression* is evaluated and its value is converted to the effective return type of the containing function by an implicit conversion. The result of the conversion becomes the result value produced by the function. For a return-by-ref, the *expression* is evaluated the same, but the result must be a reference to a *variable_reference* designated by *expression*. If the enclosing method’s return-by-ref includes `readonly`, the resulting variable is read-only.

Author: Updated text from meeting:

```suggestion
- For a return-by-value, *expression* is evaluated and its value is converted to the effective return type of the containing function by an implicit conversion. The result of the conversion becomes the result value produced by the function. For a return-by-ref, the *expression* is evaluated in the same way as a return-by-value expression, but the result must designate a *variable_reference*. If the enclosing method’s return-by-ref includes `readonly`, the resulting variable is read-only.
```
